dev/system-test/firestore.ts

```diff
@@ -2112,43 +2083,121 @@ describe('Query class', () => {
-      await collection
-        .where(
-          Filter.or(Filter.where('a', '==', 2), Filter.where('b', 'in', [2, 3]))
-        )
-        .get(),
-      'doc3',
-      'doc4',
-      'doc6'
-    );
+      // Query: a==1 || b==1 order by a.
+      // doc2 should not be included because it's missing the field 'a', and we have "orderBy a".
+      expectDocs(
+        await collection
+          .where(
+            Filter.or(Filter.where('a', '==', 1), Filter.where('b', '==', 1))
+          )
+          .orderBy('a')
+          .get(),
+        'doc1',
+        'doc4',
+        'doc5'
+      );
 
-    // a==2 || (b != 2 && b != 3)
-    // Has implicit "orderBy b"
-    expectDocs(
-      await collection
-        .where(
-          Filter.or(
-            Filter.where('a', '==', 2),
-            Filter.where('b', 'not-in', [2, 3])
+      // Query: a==1 || b==1 order by b.
+      // doc5 should not be included because it's missing the field 'b', and we have "orderBy b".
+      expectDocs(
+        await collection
+          .where(
+            Filter.or(Filter.where('a', '==', 1), Filter.where('b', '==', 1))
           )
-        )
-        .get(),
-      'doc1',
-      'doc2'
-    );
-  });
+          .orderBy('b')
+          .get(),
+        'doc1',
+        'doc2',
+        'doc4'
+      );
+
+      // Query: a>2 || b==1.
+      // This query has an implicit 'order by a'.
+      // doc2 should not be included because it's missing the field 'a'.
+      expectDocs(
+        await collection
+          .where(
+            Filter.or(Filter.where('a', '>', 2), Filter.where('b', '==', 1))
+          )
+          .get(),
+        'doc3'
+      );
+
+      // Query: a>1 || b==1 order by a order by b.
+      // doc6 should not be included because it's missing the field 'b'.
+      // doc2 should not be included because it's missing the field 'a'.
+      expectDocs(
+        await collection
+          .where(
+            Filter.or(Filter.where('a', '>', 1), Filter.where('b', '==', 1))
+          )
+          .orderBy('a')
+          .orderBy('b')
+          .get(),
+        'doc3'
+      );
+    }
+  );
+
+  // Skip this test if running against production because it results in a 'missing index' error.
+  // The Firestore Emulator, however, does serve these queries.
+  (process.env.FIRESTORE_EMULATOR_HOST === undefined ? it.skip : it)(
+    'supports OR queries with in and not-in',
+    async () => {
+      const collection = await testCollectionWithDocs({
+        doc1: {a: 1, b: 0},
+        doc2: {b: 1},
+        doc3: {a: 3, b: 2},
+        doc4: {a: 1, b: 3},
+        doc5: {a: 1},
+        doc6: {a: 2},
+      });
+
+      // Query: a==2 || b in [2, 3]
+      expectDocs(
+        await collection
+          .where(
+            Filter.or(
+              Filter.where('a', '==', 2),
+              Filter.where('b', 'in', [2, 3])
+            )
+          )
+          .get(),
+        'doc3',
+        'doc4',
+        'doc6'
+      );
```

Comment:
This one doesn't need a composite index.

so you can break it to 2 tests
`supports OR queries with in` --> doesn't need composite index (because it doesn't have inequality)
`supports OR queries with not-in` --> needs composite index.

Reply:
Split this. Thanks.